### PR TITLE
[Feature] add swin for cls

### PR DESCRIPTION
**Pull request by AllentDan**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily receiving feedbacks. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Support swin-transformer for mmcls.

## Modification

- [x] update doc
- [x] add rewritings in  `mmdeploy.codebase.mmcls`
- [x] update tests

### CODE REVIEW

**Comment by hanrui1sensetime** [comment]
Please update [mmcls.yml](https://github.com/AllentDan/mmdeploy/tree/swin-cls/tests/regression/mmcls.yml)

**Review by hanrui1sensetime — COMMENTED** [review]
I use static-224x224 config pytorch2onnx failed.(TRT 8.2.3). pls have a check.

**Comment by hanrui1sensetime** [comment]
Int8 dynamic failed

**Review by hanrui1sensetime — APPROVED** [review]
LGTM.

File: mmdeploy/codebase/mmcls/models/utils/attention.py [hunk 1]
```diff
@@ -44,3 +47,169 @@ def multiheadattention__forward__ncnn(ctx, self, qkv_input):
...
+        2, 0, 1).contiguous()  # nH, Wh*Ww, Wh*Ww
+    attn = attn + relative_position_bias.unsqueeze(0)
+
+    if mask is not None:
+        nW = mask.shape[0]
+        attn = attn.view(B_ // nW, nW, self.num_heads, N,
+                         N) + mask.unsqueeze(1).unsqueeze(0)
+        attn = attn.view(-1, self.num_heads, N, N)
+
+    means = torch.mean(attn, self.softmax.dim, keepdim=True)[0]
```
Comment (lvhan028) [inline mmdeploy/codebase/mmcls/models/utils/attention.py:80]: Why cannot we use `nn.Softmax`?

Comment (AllentDan) [reply mmdeploy/codebase/mmcls/models/utils/attention.py]: After testing, this rewriting logic seems useless. Removed.

**Review by lvhan028 — APPROVED** [review]
